frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -0,0 +1,163 @@
+import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
+import { withHandlePromise, history } from '@console/internal/components/utils';
+import { StatusBox } from '@console/internal/components/utils/status-box';
+import { StorageClassModel, PersistentVolumeClaimModel } from '@console/internal/models';
+
+import { BlockPoolModalFooter } from './modal-footer';
+import { BlockPoolStatus } from '../../block-pool/body';
+import { CephClusterKind, StoragePoolKind, OcsStorageClassResource } from '../../../types';
+import { cephClusterResource } from '../../../resources';
+import {
+  blockPoolReducer,
+  blockPoolInitialState,
+  BlockPoolActionType,
+} from '../../../utils/block-pool';
+import { POOL_PROGRESS } from '../../../constants/storage-pool-const';
+import { CephBlockPoolModel } from '../../../models';
+import { CEPH_EXTERNAL_CR_NAME, OCS_INTERNAL_CR_NAME } from '../../../constants';
+import { getSCNamesUsingPool, getPvcNames } from '../../../selectors';
+
+const DeleteBlockPoolModal = withHandlePromise((props: DeleteBlockPoolModalProps) => {
+  const { t } = useTranslation();
+  const { blockPoolConfig, cancel, close, handlePromise, inProgress } = props;
+  const poolName = blockPoolConfig?.metadata.name;
+
+  const [state, dispatch] = React.useReducer(blockPoolReducer, blockPoolInitialState);
+  const [scNames, setScNames] = React.useState<string[]>([]);
+
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+  const [scResources, scLoaded, scLoadError] = useK8sGet<ListKind<OcsStorageClassResource>>(
+    StorageClassModel,
+  );
+  const [pvcResources, pvcLoaded, pvcLoadError] = useK8sGet<ListKind<PersistentVolumeClaimKind>>(
+    PersistentVolumeClaimModel,
+  );
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
+  React.useEffect(() => {
+    // restrict pool management for default pool and external cluster
+    cephCluster?.metadata.name === CEPH_EXTERNAL_CR_NAME ||
+    blockPoolConfig?.metadata?.ownerReferences?.find(
+      (ownerReference) => ownerReference.name === OCS_INTERNAL_CR_NAME,
+    )
+      ? dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.NOTALLOWED })
+      : dispatch({
+          type: BlockPoolActionType.SET_POOL_NAME,
+          payload: poolName,
+        });
+  }, [blockPoolConfig, cephCluster, isLoaded, loadError, poolName]);
+
+  // find pvc using sc
+  const checkPvcStatus = React.useCallback(
+    (scNameList: string[]) => {
+      const pvcScNames: string[] = getPvcNames(pvcResources) || [];
+      // intersection of scNames and pvcScNames
+      const usedScNames = scNameList.filter((scName) => pvcScNames.includes(scName));
+      if (usedScNames.length) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.BOUNDED });
+        setScNames(usedScNames);
+      }
+    },
```

Comment:
Here you can move this logic to use effect below itself.
You dont need callback since `pvcResources` is a dependency there as well. Hence not providing the expected optimization.